### PR TITLE
Adds help to core modules

### PR DESCRIPTION
**Pull request by Awarua-**

### CODE REVIEW

File: core/core_modules/help.js [hunk 1]
```diff
@@ -1,5 +1,21 @@
 var figlet = require.safe('figlet'),
 
+constructHelpMessage = function(help, modules, context, prefix) {
+	for (var i = 0; i < modules.length; i++) {
+		var cmdHelp = modules[i].help.call(context, prefix);
+		for (var j = 0; j < cmdHelp.length; j++) {
+			help += '→ ' + cmdHelp[j][0] + '\n\t' + cmdHelp[j][1] + '\n';
+		}
+	}
+	return help
```
Comment (mrkno) [inline core/core_modules/help.js]: ;

File: core/core_modules/creator.js [hunk 2]
```diff
@@ -14,3 +14,7 @@ exports.run = function(api, event) {
 	api.sendMessage(creators.join('\n'), event.thread_id);
 	return false;
 };
+
+exports.help = function(commandPrefix) {
+	return [[commandPrefix + 'creator','Lists contributors to this platform']];
```
Comment (mrkno) [inline core/core_modules/creator.js]: long help

File: core/core_modules/help.js [hunk 3]
```diff
@@ -67,3 +81,7 @@ exports.run = function(api, event) {
 	api.sendPrivateMessage(help, event.thread_id, event.sender_id);
 	return false;
 };
+
+exports.help = function(commandPrefix) {
+	return [[commandPrefix + 'help','this', 'prints a short summary of all available commands'], [this.commandPrefix + 'help <query>', 'prints help for a specific module']];
```
Comment (mrkno) [inline core/core_modules/help.js]: 'this' -> 'displays this help'

File: core/modules.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ exports.loadCoreModule = function(platform, module) {
         index = Object.keys(require.cache).indexOf(fp),
         m = index !== -1 ? require.reload(fp) : require.once(fp);
     m.platform = exports;
-    m.name = module;
+    m.name = module.split(".")[0];
```
Comment (mrkno) [inline core/modules.js]: Why?

Comment (Awarua-) [inline core/modules.js]: To remove the .js on the end of the name, other wise !help creator is not recognised, instead !help creator.js would be required, can store extra variable if you prefer, just made the name inline with a normal module

Comment (mrkno) [inline core/modules.js]: ok

**Pull request merged**